### DISCUSSION TITLE
[record] too many reordered frames

### DISCUSSION THREAD

**Discussion by svk23** [bluenviron/mediamtx#3076] [Questions]
### Question

I have a NVC4000, which is a VCA / UDP Technology H.264 Network Video Encoder. I am trying to use MediaMTX version 1.5.1 to record the RTSP stream to MpegTS files, but the server continuously logs errors about too many reordered frames:

```
2024/02/27 14:15:16 INF [path test] [record] recording 1 track (H264)
2024/02/27 14:15:16 INF [path test] [RTSP source] ready: 1 track (H264)
2024/02/27 14:15:16 ERR [path test] [record] too many reordered frames (15126)
2024/02/27 14:15:18 INF [path test] [record] recording 1 track (H264)
2024/02/27 14:15:18 ERR [path test] [record] too many reordered frames (15106)
2024/02/27 14:15:20 INF [path test] [record] recording 1 track (H264)
2024/02/27 14:15:20 ERR [path test] [record] too many reordered frames (15086)
2024/02/27 14:15:22 INF [path test] [record] recording 1 track (H264)
2024/02/27 14:15:24 ERR [path test] [record] too many reordered frames (15046)
2024/02/27 14:15:26 INF [path test] [record] recording 1 track (H264)
2024/02/27 14:15:26 ERR [path test] [record] too many reordered frames (15026)
2024/02/27 14:15:28 INF [path test] [record] recording 1 track (H264)
2024/02/27 14:15:30 ERR [path test] [record] too many reordered frames (14986)
```

The server seems to record about a single frame before restarting. I can't figure out if this is a bug (perhaps the wrong poc is read) or if the encoder is returning invalid data. Is there anything I can do to find out what is going on before filing this as a bug?

**Comment by aler9** [comment]
Hello, open a bug report and provide a network dump. The two options you mentioned (wrong reading of POC and invalid data) may be both true. You can also try to read the original stream with FFmpeg and check whether there are some errors:

```
ffmpeg -i rtsp://original-stream -f null -
```

**Reply by quanglinh2909** [reply]
i also got the same error when saving video from an rtsp stream, even though i set recordSegmentDuration value as 5 minutes but the video i get is small segments from 0.1 to 2 seconds.

![image](https://github.com/bluenviron/mediamtx/assets/69236990/b89e32dd-56d6-4e3a-8d2b-281502fee42e)
![image](https://github.com/bluenviron/mediamtx/assets/69236990/60a6da90-62c8-4a61-ab21-76930d0e3d4d)

**Reply by aler9** [reply]
it's impossible to understand what the problem is without a network dump or a video that allows to replicate the issue. Open a dedicated bug report or question.
